guides/keras_cv/simsiam-with-kerascv.py

```diff
@@ -0,0 +1,604 @@
+Using KerasCV, it is trivial to construct an augmenter that performs as the one
+described in the original SimSiam paper.  Lets do that below.
+"""
+
+target_size = (96, 96)
+crop_area_factor = (0.08, 1)
+aspect_ratio_factor = (3 / 4, 4 / 3)
+grayscale_rate = 0.2
+color_jitter_rate = 0.8
+brightness_factor = 0.2
+contrast_factor = 0.8
+saturation_factor = (0.3, 0.7)
+hue_factor = 0.2
+
+augmenter = keras_cv.layers.Augmenter(
+    [
+        cv_layers.RandomFlip("horizontal"),
+        cv_layers.RandomCropAndResize(
+            target_size,
+            crop_area_factor=crop_area_factor,
+            aspect_ratio_factor=aspect_ratio_factor,
+        ),
+        cv_layers.MaybeApply(
+            cv_layers.Grayscale(output_channels=3), rate=grayscale_rate
+        ),
+        cv_layers.MaybeApply(
+            cv_layers.RandomColorJitter(
+                value_range=(0, 255),
+                brightness_factor=brightness_factor,
+                contrast_factor=contrast_factor,
+                saturation_factor=saturation_factor,
+                hue_factor=hue_factor,
+            ),
+            rate=color_jitter_rate,
+        ),
+    ],
+)
+
+"""
+Next, lets pass our images through this pipeline.
+Note that KerasCV supports batched augmentation, so batching before
+augmentation dramatically improves performance
+
+"""
+
+
+@tf.function()
+def process(img):
+    return augmenter(img), augmenter(img)
+
+
+train_ds = train_ds.repeat()
+train_ds = train_ds.shuffle(1024)
+train_ds = train_ds.batch(BATCH_SIZE)
+train_ds = train_ds.map(process, num_parallel_calls=tf.data.AUTOTUNE)
+train_ds = train_ds.prefetch(tf.data.AUTOTUNE)
+
+val_ds = tf.data.Dataset.from_tensor_slices(x_val)
+val_ds = val_ds.repeat()
+val_ds = val_ds.shuffle(1024)
+val_ds = val_ds.batch(BATCH_SIZE)
+val_ds = val_ds.map(process, num_parallel_calls=tf.data.AUTOTUNE)
+val_ds = val_ds.prefetch(tf.data.AUTOTUNE)
+
+print("train_ds", train_ds)
+print("val_ds", val_ds)
+
+"""
+Lets visualize our pairs using the `tfsim.visualization` utility package.
+"""
+display_imgs = next(train_ds.as_numpy_iterator())
+max_pixel = np.max([display_imgs[0].max(), display_imgs[1].max()])
+min_pixel = np.min([display_imgs[0].min(), display_imgs[1].min()])
+
+tfsim.visualization.visualize_views(
+    views=display_imgs,
+    num_imgs=16,
+    views_per_col=8,
+    max_pixel_value=max_pixel,
+    min_pixel_value=min_pixel,
+)
+
+"""
+## Model Creation
+
+Now that our data and augmentation pipeline is setup, we can move on to
+constructing the contrastive learning pipeline.  First, lets produce a backbone.
+For this task, we will use a KerasCV ResNet18 model as the backbone.
+"""
+
+
+def get_backbone(input_shape):
+    inputs = layers.Input(shape=input_shape)
+    x = inputs
+    x = keras_cv.models.ResNet18(
+        input_shape=input_shape,
+        include_rescaling=True,
+        include_top=False,
+    )(x)
+    x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
```

Comment:
We should be able to just pass `pooling="avg"` to the ResNet18 call to add this.